nullaway/src/main/java/com/uber/nullaway/handlers/StreamNullabilityPropagator.java

```diff
@@ -113,9 +118,16 @@ class StreamNullabilityPropagator extends BaseNoOpHandler {
   private final Map<MethodInvocationTree, Tree> observableCallToInnerMethodOrLambda =
       new LinkedHashMap<>();
 
-  // Map from map method (or lambda) to corresponding previous filter method (e.g. B.apply =>
-  // A.filter)
-  private final Map<Tree, MaplikeToFilterInstanceRecord> mapToFilterMap = new LinkedHashMap<>();
+  // Maps collect calls in the observable call chain to the relevant inner methods or lambdas.
+  // We need a Multimap here since there may be multiple relevant methods / lambdas.
+  // E.g.: stream.filter(...).collect(Collectors.toMap(l1, l2)) => {l1,l2}
+  private final Multimap<MethodInvocationTree, Tree> collectCallToInnerMethodsOrLambdas =
+      LinkedHashMultimap.create();
+
+  // Map from map or collect method (or lambda) to corresponding previous filter method (e.g.
+  // B.apply => A.filter)
```

Comment:
Probably need an example here for the collect case, specially if we update the global docs above.

Reply:
1b57a96